### PR TITLE
Add BlockHeader tests

### PR DESCRIPTION
**Pull request by 0xB10C**
To create the the two `BlockHeader::work()` test vectors I used https://play.golang.org/p/N4_0K87GZfP. Checking these against Bitcoin Core can only be done by calling [`GetBlockProof()`](https://github.com/bitcoin/bitcoin/blob/c7ad94428ab6f54661d7a5441e1fdd0ebf034903/src/chain.cpp#L122) in `chain.cpp`. The per-block work is not exposed via RPC (only the chainwork).

Fixes https://github.com/rust-bitcoin/rust-bitcoin/issues/462.

### CODE REVIEW

File: src/blockdata/block.rs [hunk 1]
```diff
@@ -388,6 +397,9 @@ mod tests {
         assert_eq!(real_decode.header.time, 1472004949);
         assert_eq!(real_decode.header.bits, 0x1a06d450);
         assert_eq!(real_decode.header.nonce, 1879759182);
+        assert_eq!(real_decode.header.work(), work);
+        assert_eq!(real_decode.header.validate_pow(&real_decode.header.target()).unwrap(), ());
+        assert_eq!(real_decode.header.difficulty(Network::Testnet), 2456598);
```
Comment (0xB10C) [inline src/blockdata/block.rs:402]:
```
bitcoin-cli -testnet getblockheader 000000000000045e0b1660b6445b5e5c5ab63c9a4f956be7e1e69be04fa4497b | jq .difficulty
```

File: src/blockdata/block.rs [hunk 2]
```diff
@@ -358,6 +363,9 @@ mod tests {
         assert_eq!(real_decode.header.time, 1231965655);
         assert_eq!(real_decode.header.bits, 486604799);
         assert_eq!(real_decode.header.nonce, 2067413810);
+        assert_eq!(real_decode.header.work(), work);
+        assert_eq!(real_decode.header.validate_pow(&real_decode.header.target()).unwrap(), ());
+        assert_eq!(real_decode.header.difficulty(Network::Bitcoin), 1);
```
Comment (0xB10C) [inline src/blockdata/block.rs:368]: bitcoin-cli getblockheader 00000000b0c5a240b2a61d2e75692224efd4cbecdf6eaf4cc2cf477ca7c270e7 | jq .difficulty

**Review by sgeisler — APPROVED** [review]
Awesome, thx!

File: src/blockdata/block.rs [hunk 3]
```diff
@@ -393,6 +395,7 @@ mod tests {
         assert_eq!(real_decode.header.bits, 0x1a06d450);
         assert_eq!(real_decode.header.nonce, 1879759182);
         assert_eq!(real_decode.header.work(), work);
+        assert_eq!(real_decode.header.validate_pow(&real_decode.header.target()).unwrap(), ());
```
Comment (stevenroose) [inline src/blockdata/block.rs:401]: nit: The use of assert_eq here is a bit pointless, but well.

**Review by stevenroose — APPROVED** [review]
LGTM, thanks

tACK b7caf0110f2ac86d9040564d72c8ce156cf41656

**Review by apoelstra — APPROVED** [review]
ack b7caf0110f2ac86d9040564d72c8ce156cf41656

**Pull request merged**